src/lib.rs

```diff
@@ -503,23 +517,50 @@ impl Contract {
         handle: CommunityHandle,
         #[allow(unused_mut)] mut community: Community,
     ) {
-        let target_community = self
+        let _ = self
             .get_editable_community(&handle)
             .expect("Only community admins and hub moderators can configure communities");
 
         community.validate();
         community.set_default_admin();
 
-        if target_community.handle == community.handle {
-            self.communities.insert(&target_community.handle, &community);
-        } else {
-            require!(
-                self.communities.get(&community.handle).is_none(),
-                "Community handle is already taken"
-            );
-            self.communities.remove(&target_community.handle);
-            self.communities.insert(&community.handle, &community);
-        }
+        require!(community.handle == handle, "Community handle cannot be changed");
+        require!(env::prepaid_gas() >= UPDATE_COMMUNITY_GAS, "Require at least 30 Tgas");
+        self.communities.insert(&handle, &community);
+        ext_social_db::ext(SOCIAL_DB.parse().unwrap()).with_unused_gas_weight(1).set(json!({
+            format!("{}.{}", community.handle, DEVHUB_COMMUNITY_FACTORY)
+            : {
+                "profile": {
+                    "name": community.name,
+                    "image": {
+                        "url": community.logo_url,
+                    }
+                }
+            }
+        }));
+    }
+
+    pub fn add_community_announcement(
+        &mut self,
+        handle: CommunityHandle,
+        announcement_post: String,
+    ) {
+        let _ = self
+            .get_editable_community(&handle)
+            .expect("Only community admins and hub moderators can create announcement");
+
+        require!(env::prepaid_gas() >= ADD_COMMUNITY_ANNOUNCEMENT_GAS, "Require at least 30 Tgas");
+        ext_social_db::ext(SOCIAL_DB.parse().unwrap()).with_unused_gas_weight(1).set(json!({
+            format!("{}.{}", handle, DEVHUB_COMMUNITY_FACTORY)
+            : {
+                "post": {
+                    "main": to_string(&json!({"type": "md", "text": announcement_post})).unwrap(),
+                    "index": {
+                        "post": "{\"key\":\"main\",\"value\":{\"type\":\"md\"}}"
```

Comment:
It would be nice if this "index" could also be passed in from the function call, for more flexibility and this will be very helpful when implementing discussions.

With the two above comments in mind, proposal to rename this function to "add_community_post", where it takes a community handle and the whole object that would otherwise be passed to Social.set

From the front end, I imagine calling it like:
```
Near.call("devgovgigs.near", "add_community_post", { handle: "devhub-test", content: JSON.stringify({
    "post": {
      "main": "{\"type\":\"md\",\"text\":\"\",\"image\":{\"ipfs_cid\":\"bafkreigf7dza2byuwougbvhwodr2t5jofmln4ivqgixs7czr27kjanq5ge\"}}"
    },
    "index": {
      "post": "{"key\":\"main\",\"value\":{\"type\":\"md\"}}"
    }
  }) });
  ```

Reply:
This would require frontend take care and ensure the format passing to it is valid, but should be fine as only community admins and moderators can call add_community_announcement